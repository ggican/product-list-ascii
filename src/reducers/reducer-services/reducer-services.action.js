import service from "../../services";
import localStorage from "../../utils/local-storage";

const testing = (dispatch, value) => {
    dispatch(value);
};
const login = (dispatch, value) => {
    localStorage().save("auth", value.data);
    dispatch(value);
};

const clearData = (dispatch, value) => {
    dispatch(value);
};
const productPurchase = (dispatch, value) => {
    dispatch(value);
};

const fetch = async (dispatch, value) => {
    const result = await service(value);
    if (result) {
        dispatch({
            reducer: value.reducer,
            group: value.group,
            key: value.key,
            data: result.response,
            message: result.message,
            type: result.type,
            isError: result.error,
            isSuccess: result.success,
        });
    }
};

const productService = async (dispatch, value) => {
    Promise.all(
        value.map(item => {
            return service(item);
        }),
    ).then(responseAll => {
        console.log(responseAll);
        // responseAll.map(result => {
        //     dispatch({
        //         headers: result.headers,
        //         response: result.response,
        //         status: result.status,
        //         type: result.type,
        //         message: result.message && result.message[result.status],
        //         reducer: result.reducer,
        //         group: result.group,
        //         key: result.key,
        //         params: value.params,
        //     });
        // });
    });
    // const result = await service(value);
    // if (result) {
    //     dispatch({
    //         reducer: value.reducer,
    //         group: value.group,
    //         key: value.key,
    //         data: result.response,
    //         message: result.message,
    //         type: result.type,
    //         isError: result.error,
    //         isSuccess: result.success,
    //     });
    // }
};

const servicesAction = dispatch => ({
    reduxFetch: value => {
        fetch(dispatch, value);
    },
    productService: value => {
        productService(dispatch, value);
    },
    reduxClear: value => {
        clearData(dispatch, value);
    },
    reduxTesting: value => {
        testing(dispatch, value);
    },
    reduxLogin: value => {
        login(dispatch, value);
    },
    reduxProductPurchase: value => {
        productPurchase(dispatch, value);
    },
});

export default servicesAction;
