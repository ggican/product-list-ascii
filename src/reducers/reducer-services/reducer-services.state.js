const serviceState = {
    user: {
        auth: false,
    },
    product: {
        list: false,
    },
};

export default serviceState;
