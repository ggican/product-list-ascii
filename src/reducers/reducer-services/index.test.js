import { servicesRedux, servicesState } from "./reducer-services.redux";

describe("Reducer Services Component", () => {
    const state = servicesState;
    const result = {
        user: {
            auth: false,
        },
        product: {
            list: false,
        },
    };
    beforeEach(() => {});

    it("SUCCESS servicesRedux Snapshot servicesRedux", () => {
        const actionTest = {
            reducer: "reducer",
            group: "product",
            key: "list",
            data: ["cool"],
            message: {
                message: "cool its me",
            },
            type: 200,
            isError: false,
            isSuccess: true,
        };

        expect(servicesRedux(servicesState, actionTest)).toMatchSnapshot();
    });
    it("SUCCESS servicesState  Snapshot servicesState", () => {
        expect(servicesState).toMatchSnapshot();
    });
    it("SUCCESS servicesRedux type 200 function servicesRedux", () => {
        const actionTest = {
            reducer: "reducer",
            group: "product",
            key: "list",
            data: ["cool"],
            message: {
                message: "cool its me",
            },
            type: 200,
            isError: false,
            isSuccess: true,
        };

        expect(servicesRedux(state, actionTest)).toEqual({
            ...result,
            product: {
                list: actionTest,
            },
        });
    });

    it("SUCCESS servicesRedux type 201 function servicesRedux", () => {
        const actionTest = {
            reducer: "reducer",
            group: "product",
            key: "list",
            data: ["cool"],
            message: {
                message: "cool its me",
            },
            type: 201,
            isError: false,
            isSuccess: true,
        };
        expect(servicesRedux(state, actionTest)).toEqual({
            ...result,
            product: {
                list: actionTest,
            },
        });
    });

    it("SUCCESS servicesRedux type 201 function servicesRedux", () => {
        const actionTest = {
            reducer: "reducer",
            group: "product",
            key: "list",
            data: ["cool"],
            message: {
                message: "cool its me",
            },
            type: 204,
            isError: false,
            isSuccess: true,
        };
        expect(servicesRedux(state, actionTest)).toEqual({
            ...result,
            product: {
                list: actionTest,
            },
        });
    });

    it("SUCCESS servicesRedux type 201 function servicesRedux", () => {
        const actionTest = {
            reducer: "reducer",
            group: "product",
            key: "list",
            data: ["cool"],
            message: {
                message: "cool its me",
            },
            type: 400,
            isError: true,
            isSuccess: false,
        };
        expect(servicesRedux(state, actionTest)).toEqual({
            ...result,
            product: {
                list: actionTest,
            },
        });
    });

    it("SUCCESS servicesRedux type 401 function servicesRedux", () => {
        const actionTest = {
            reducer: "reducer",
            group: "product",
            key: "list",
            data: ["cool"],
            message: {
                message: "cool its me",
            },
            type: 401,
            isError: true,
            isSuccess: false,
        };
        expect(servicesRedux(state, actionTest)).toEqual({
            ...result,
            product: {
                list: actionTest,
            },
        });
    });

    it("SUCCESS servicesRedux type 404 function servicesRedux", () => {
        const actionTest = {
            reducer: "reducer",
            group: "product",
            key: "list",
            data: ["cool"],
            message: {
                message: "cool its me",
            },
            type: 404,
            isError: true,
            isSuccess: false,
        };
        expect(servicesRedux(state, actionTest)).toEqual({
            ...result,
            product: {
                list: actionTest,
            },
        });
    });

    it("SUCCESS servicesRedux type 409 function servicesRedux", () => {
        const actionTest = {
            reducer: "reducer",
            group: "product",
            key: "list",
            data: ["cool"],
            message: {
                message: "cool its me",
            },
            type: 409,
            isError: true,
            isSuccess: false,
        };
        expect(servicesRedux(state, actionTest)).toEqual({
            ...result,
            product: {
                list: actionTest,
            },
        });
    });

    it("SUCCESS servicesRedux type 500 function servicesRedux", () => {
        const actionTest = {
            reducer: "reducer",
            group: "product",
            key: "list",
            data: ["cool"],
            message: {
                message: "cool its me",
            },
            type: 500,
            isError: true,
            isSuccess: false,
        };
        expect(servicesRedux(state, actionTest)).toEqual({
            ...result,
            product: {
                list: actionTest,
            },
        });
    });
    it("SUCCESS servicesRedux type CLEAR function servicesRedux", () => {
        const actionTest = {
            reducer: "reducer",
            group: "product",
            key: "list",
            defaultData: {
                isDefault: true,
            },
            type: "CLEAR",
        };
        expect(servicesRedux(state, actionTest)).toEqual({
            ...result,
            product: {
                list: {
                    isDefault: true,
                },
            },
        });
    });
    it("SUCCESS servicesRedux type DEFAULT function servicesRedux", () => {
        const actionTest = {
            type: false,
        };
        expect(servicesRedux(state, actionTest)).toEqual({
            ...result,
            product: {
                list: false,
            },
        });
    });
});
