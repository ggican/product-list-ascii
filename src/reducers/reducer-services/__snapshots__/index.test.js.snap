// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Reducer Services Component SUCCESS servicesRedux Snapshot servicesRedux 1`] = `
Object {
  "product": Object {
    "list": Object {
      "data": Array [
        "cool",
      ],
      "group": "product",
      "isError": false,
      "isSuccess": true,
      "key": "list",
      "message": Object {
        "message": "cool its me",
      },
      "reducer": "reducer",
      "type": 200,
    },
  },
  "user": Object {
    "auth": false,
  },
}
`;

exports[`Reducer Services Component SUCCESS servicesState  Snapshot servicesState 1`] = `
Object {
  "product": Object {
    "list": false,
  },
  "user": Object {
    "auth": false,
  },
}
`;
